dbusmock/mockobject.py

```diff
@@ -326,6 +356,21 @@ def Reset(self) -> None:
 
         objects[self.path] = self
 
+    def AddServiceMethod(self, interface, name, method) -> None:
+        '''Add service decorated method
+
+        interface: D-Bus interface to add this to. For convenience you can
+                   specify '' here to add the method to the object's main
+                   interface (as specified on construction).
+        name: Name of the method
+        method: A method decorated using @dbus.service.method
+        '''
+        # pylint: disable=protected-access
+        self.AddMethod(interface, name,
+                       method._dbus_in_signature,
+                       method._dbus_out_signature,
```

Comment:
Maybe we can protect here checking whether we're handling a function and if hasattr such values? Raising otherwise?

If doing this though, we could just maybe avoid adding another function and just add this as a special case of `AddMethod`.... Actually this will also solve another problem we have with decoration arguments so, could you instead pick from my branch?

https://github.com/3v1n0/python-dbusmock/commits/systemd-unit

Reply:
I'll wait for @martinpitt to know whether he wants the regular AddMethod to handle this, and if so, I think the AddServiceMethod is a bit unnecessary.

Reply:
In any case we can discuss that prior change in another PR, but I guess it's quite needed, as as explained many decorator parameters won't work otherwise.

Reply:
Honestly this feels a bit weird and bolted-on. MockObject is a D-Bus object itself, its API is via D-Bus. You can't call AddServiceMethod() through that. There is two ways to add a mocked method:

 1. through `AddMethod()` with the code as string; this works for any programming language (even at the shell), and is meant for dynamically shaping mocks in a project's unit tests.
 2. declaring a proper D-Bus function in the template [like here](https://github.com/martinpitt/python-dbusmock/blob/master/dbusmock/templates/logind.py#L82); this is for writing non-trivial methods in templates.

Why do you need this "hybrid" and indirect approach?

Reply:
No longer relevant, AddServiceMethod dropped as it was no longer needed.